### DISCUSSION TITLE
Reproducibility features

### DISCUSSION THREAD

**Discussion by EvanZ** [catboost/catboost#3077] [General]
We are trying to make our models deterministic for testing purposes, and while I know there are certain things (such as using a GPU) that can make models non-deterministic, we're finding that it's very unpredictable. For example, models using poisson loss seem to be deterministic given the same data and seed when run on a CPU (even multicore). But we can't seem to make classifiers deterministic. I'm hoping some here may have gone down this rabbit hole and can confirm one way or the other whether truly deterministic models can be created (ie reproducible predictions given the same training data, hyperparameters, etc).
